### PR TITLE
Add input correction

### PR DESCRIPTION
**Pull request by schillic**
The code is almost identical with `correction_hull`, so I shared the code and renamed the file.

### CODE REVIEW

File: src/correction_matrices.jl [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+function _correction_loop!(F, A::IntervalMatrix{T}, Aⁱ, t, p) where {T}
+    i! = 1
+    tⁱ = t
+    @inbounds for i in 2:p
+        tⁱ *= t
+        left = (one(T) / i^(i/(i-1)) - one(T) / i^(1/(i-1))) * tⁱ
+        itv = Interval(left, zero(T))
+        Aⁱ = Aⁱ * A
+        i! *= i
+        F += itv * Aⁱ * (1/i!)
```
Comment (mforets) [inline src/correction_matrices.jl]:
```suggestion
        F += itv * Aⁱ * (one(T)/i!)
```
or `F += itv * Aⁱ / i!` ? (idk if it makes any difference)

Comment (mforets) [reply src/correction_matrices.jl]: (`itv * Aⁱ / i!` reads better imo, but either way is fine)

Comment (schillic) [reply src/correction_matrices.jl]: This was old code but I agree :+1:

File: src/correction_matrices.jl [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+end
+
+# inputs:
+# F:  interval matrix sum (modified)
+# A:  interval matrix
+# Aⁱ: interval matrix factor in iteration i to start with; examples:
+#     if Aⁱ == A, then this function will multiply with Aⁱ
+#     if Aⁱ == id, then this function will multiply with Aⁱ⁻¹
+# t:  time step
+# p:  Taylor-approximation order
```
Comment (mforets) [inline src/correction_matrices.jl]: i would add that this computes

```math
 F = \\sum_{i=2}^p [...., 0] * t^i * (Aⁱ * A^i)
``
but lets leave the doc for another PR if you are fine with that

Comment (schillic) [reply src/correction_matrices.jl]: I added this line and we can improve it later.

File: src/correction_matrices.jl [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    i! = 1
+    tⁱ = t
+    @inbounds for i in 2:p
+        tⁱ *= t
+        left = (one(T) / i^(i/(i-1)) - one(T) / i^(1/(i-1))) * tⁱ
+        itv = Interval(left, zero(T))
+        Aⁱ = Aⁱ * A
+        i! *= i
+        F += itv * Aⁱ / i!
+    end
```
Comment (mforets) [inline src/correction_matrices.jl]:
```suggestion
    return F
    end
```
this is optional but i prefer that the main object of the function is returned
